sklearn/utils/seq_dataset.pxd

```diff
@@ -25,10 +33,6 @@ cdef class ArrayDataset(SequentialDataset):
     cdef int *feature_indices_ptr
     cdef double *sample_weight_data
 
-    cdef void next(self, double **x_data_ptr, int **x_ind_ptr,
-                   int *nnz, double *y, double *sample_weight) nogil
-
```

Comment:
I don't know, is it required? @dsullivan7 ?
Do we also need to add on both `ArrayDataset` and `CSRDataset`
`cdef void _sample(self, double **x_data_ptr, int **x_ind_ptr, int *nnz, double *y, double *sample_weight, int current_index) nogil` ?